processing/src/main/java/io/druid/query/aggregation/BufferAggregator.java

```diff
@@ -110,4 +110,16 @@
    * Release any resources used by the aggregator
    */
   void close();
+
+  /**
+   * Relocates any cached objects.
+   * <b>Implementations must not change the position, limit or mark of the given buffer</b>
+   *
+   * @param oldPostition old position of an item in old ByteBuffer.
+   * @param newPosition  new position of an item in new ByteBuffer.
+   * @param newBuffer    ByteBuffer to be used.
+   */
+  default void relocate(int oldPostition, int newPosition, ByteBuffer newBuffer)
```

Comment:
It will be difficult to put `BufferGrouper` functionality as a code comment in `BufferAggregator`, I'll try to put more details without bringing in the `BufferGrouper` complexities.
I disagree with putting examples in java doc and also don't see for other methods.

`Relocate` is not a hot method(only called when `BufferGrouper` grows) and I think keeping the existing structure should be okay.